test/test_rpc.py

```diff
@@ -1,64 +1,201 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import functools
 import sys
 
 import torch
 import torch.distributed as dist
-
-from common_distributed import MultiProcessTestCase
+from common_distributed import MultiProcessTestCase, skip_for_known_issues
 from common_utils import load_tests, run_tests
 
 
+class WorkerContext(object):
```

Comment:
Perhaps I should write a C++ class for it.

And it should eventually evolve into a WorkerDiscoveryService Interface and integrate with PET (PyTorch Elastic Trainer).

Reply:
Good point! I think we can keep this here for now. If we do want to evolve this into `WorkerDiscoveryService` we might need to have discussion on the features and API and make sure it works for different RPC backend.